packages/react/src/utilities/selection/SelectionZone.tsx

```diff
@@ -708,15 +751,20 @@ export class SelectionZone extends React.Component<ISelectionZoneProps, ISelecti
     }
   }
 
-  private _clearAndSelectIndex(index: number): void {
+  private _clearAndSelectIndex(index: number, span?: number): void {
     const { selection, selectionClearedOnSurfaceClick = true } = this.props;
-    const isAlreadySingleSelected = selection.getSelectedCount() === 1 && selection.isIndexSelected(index);
+    const isAlreadySingleSelected =
+      (span === undefined || span === 1) && selection.getSelectedCount() === 1 && selection.isIndexSelected(index);
 
     if (!isAlreadySingleSelected && selectionClearedOnSurfaceClick) {
       const isModal = selection.isModal && selection.isModal();
       selection.setChangeEvents(false);
       selection.setAllSelected(false);
-      selection.setIndexSelected(index, true, true);
+      if (span !== undefined) {
+        selection.setRangeSelected(index, span, true, true);
+      } else {
+        selection.setIndexSelected(index, true, true);
```

Comment:
This suggestion may actually make the code more awkward to read but what do you think about updating the code to treat single indexes as a span of 1? To make this change you would keep `span` an optional parameter but default it to `1`. Then you could eliminate all the `if (span !== undefined)` checks and just use the `range` variants of the selection methods.

The `range` selection methods are little more complicated and hence probably somewhat slower so we could perhaps have conditional logic to handle spans of 1 in these range methods.

Reply:
There is some logic, notably handling of selection-on-focus, which needs to know if the given element has a `span` attribute *at all*. For those elements, we don't want to automatically update selection. So I still would need some indicator for whether or not a span is in play, even if it is `1`. For that reason, I think leveraging `undefined` is a good idea. I will make the change to handle `NaN` specifically, though.